WordPress/Classes/Services/ReaderPostStreamService.swift

```diff
@@ -27,44 +29,41 @@ extension ReaderPostService {
 
                 if isFirstPage {
                     self.pageNumber = 1
-                    self.removePosts(forTopic: readerTopic)
+                    self.removePosts(forTopic: readerTopic, in: context)
                 } else {
                     self.pageNumber += 1
                 }
 
                 posts.enumerated().forEach { index, remotePost in
-                    let post = ReaderPost.createOrReplace(fromRemotePost: remotePost, for: readerTopic, context: self.managedObjectContext)
+                    let post = ReaderPost.createOrReplace(fromRemotePost: remotePost, for: readerTopic, context: context)
                     // To keep the API order
                     post?.sortRank = NSNumber(value: Date().timeIntervalSinceReferenceDate - Double(((self.pageNumber * Constants.paginationMultiplier) + index)))
                 }
 
                 // Clean up
-                self.deletePostsInExcessOfMaxAllowed(for: readerTopic)
-                self.deletePostsFromBlockedSites()
-
-                ContextManager.shared.save(self.managedObjectContext) {
-                    let hasMore = pageHandle != nil
-                    success(posts.count, hasMore)
-                }
-
+                let serivce = ReaderPostService(managedObjectContext: context)
+                serivce.deletePostsInExcessOfMaxAllowed(for: readerTopic)
+                serivce.deletePostsFromBlockedSites()
```

Comment:
Is it guaranteed that `ReaderPostService` will be retained during this operation? Do we not need to keep a reference to it?

Reply:
By keeping a reference, do you mean saving it a stored property? If that's the case, I don't think we need to do that. With changes in this PR, `ReaderPostSerivce` is now stateless, like almost all other "service" types. So, it can be used in a create-and-discard way, like almost all other "service" types.